sls-versions/src/main/java/com/palantir/sls/versions/OrderableSlsVersion.java

```diff
@@ -54,37 +53,35 @@ public static OrderableSlsVersion valueOf(String value) {
             return Optional.empty();
         }
 
-        SlsVersionType type = getTypeOrNull(value);
-        if (type == null) {
+        SlsVersionType finalType = ORDERED_VERSION_TYPES[0];
+        Matcher groups = null;
+        for (SlsVersionType type : ORDERED_VERSION_TYPES) {
+            Matcher maybeGroups = type.getParser().tryParse(value);
+            if (maybeGroups != null) {
+                finalType = type;
+                groups = maybeGroups;
+                break;
```

Comment:
Might be cleaner to split method in half, and replace these 3 lines with `return callSecondHalf(type, maybeGroups);`

Reply:
great point - reads much nicer